packages/team-management/src/components/TeamsAddEdit/index.tsx

```diff
@@ -90,6 +91,73 @@ export const defaultProps = {
   opensrpBaseURL: '',
 };
 
+/**
+ * function to fetch a paginated practitioner object
+ *
+ * @param opensrpBaseURL - OpenSRP API base URL
+ * @param practitionersEndpoint - OpenSRP practitioners endpoint
+ * @param pageNumber - paginated page number
+ * @param pageSize - paginated page size
+ * @param errorNotification - the error message to show for a failed request
+ * @returns {Promise<Practitioner[]>} an array of practitioners
+ */
+async function fetchPractitioners(
+  opensrpBaseURL: string,
+  practitionersEndpoint: string,
+  pageNumber: number,
+  pageSize: number,
+  errorNotification: string = lang.ERROR_OCCURRED
+): Promise<Practitioner[]> {
+  const composePaginatedEndpoint = `${practitionersEndpoint}?pageNumber=${pageNumber}&pageSize=${pageSize}`;
+  const serve = new OpenSRPService(composePaginatedEndpoint, opensrpBaseURL);
+  try {
+    const practitioners: Practitioner[] = await serve.list();
+    return practitioners;
+  } catch (_) {
+    sendErrorNotification(errorNotification);
+    return [];
+  }
+}
+
+/**
+ * function to fetch paginated practitioners resource recursively
+ *
+ * @param opensrpBaseURL - OpenSRP API base URL
+ * @param practitionersCountEndpoint - OpenSRP practitioners count endpoint
+ * @param practitionersEndpoint - OpenSRP practitioners endpoint
+ * @returns {Promise<Practitioner[]>} - an array of all practitioners in a paginated endpoint
+ */
+async function fetchPractitionersRecursively(
+  opensrpBaseURL: string,
+  practitionersCountEndpoint: string,
+  practitionersEndpoint: string
+): Promise<Practitioner[]> {
+  // practitioner array to fill recursively
+  const allPractitioners: Practitioner[] = [];
+
+  // get the total number of practitioners
+  const serve = new OpenSRPService(practitionersCountEndpoint, opensrpBaseURL);
+  const practitionerCount: number = await serve.list();
+
+  const pageSize = 1000;
+
+  // get the maximum possible page numbers
+  const maxPageNo = Math.ceil(practitionerCount / pageSize);
+
+  // add practitioners recursively based on page number
+  for (let pageNumber = 1; pageNumber <= maxPageNo; pageNumber++) {
+    const practitioners = await fetchPractitioners(
+      opensrpBaseURL,
+      practitionersEndpoint,
+      pageNumber,
+      pageSize
+    );
+    allPractitioners.push(...practitioners);
+  }
```

Comment:
are the requests here blocking, if yes, would it be better to async-ly call them.

Reply:
@p-netm any Idea how to do that?

Reply:
could Promise.all work here?

Reply:
I'll give it a try

Reply:
aff169d3df31d951cd73715adc4604ec0f6bfbe2